contracts/integrations/test/framework-unit-tests/function_assertion_test.ts

```diff
@@ -30,10 +23,10 @@ blockchainTests.resets('FunctionAssertion Unit Tests', env => {
         it('should call the before function with the provided arguments', async () => {
             let sideEffectTarget = ZERO_AMOUNT;
             const assertion = new FunctionAssertion(exampleContract.returnInteger, {
-                before: async (input: BigNumber) => {
+                before: async (_input: BigNumber) => {
                     sideEffectTarget = randomInput;
                 },
-                after: async (beforeInfo: any, result: Result, input: BigNumber) => {},
+                after: async (_beforeInfo: any, _result: Result, _input: BigNumber) => null,
```

Comment:
We seem to be doing this a lot. Maybe it's out of the scope of this PR, but why can't `FunctionAssertion` just take a `Partial<Condition>` and do something like `this.condition = {after: _.noop, before: _.noop, ...condition }` in its constructor?

Reply:
I suggested something along these lines in @jalextowle 's PR but iirc we can't because of the respective function signatures of `before` and `after` ?

Reply:
We might be able to find a way to seed `beforeInfo` with a default, but as of now, I think that is the main barrier to allowing optionality.